plugins/node/opentelemetry-instrumentation-pg/src/instrumentation.ts

```diff
@@ -367,17 +351,17 @@ export class PgInstrumentation extends InstrumentationBase {
         }
 
         // setup span
-        const span = plugin.tracer.startSpan(`${PG_POOL_COMPONENT}.connect`, {
+        const span = plugin.tracer.startSpan('pg-pool.connect', {
           kind: SpanKind.CLIENT,
-          attributes: {
-            [SemanticAttributes.DB_SYSTEM]: DbSystemValues.POSTGRESQL,
-            ...utils.getSemanticAttributesFromConnection(this.options),
-            [AttributeNames.IDLE_TIMEOUT_MILLIS]:
-              this.options.idleTimeoutMillis,
-            [AttributeNames.MAX_CLIENT]: this.options.maxClient,
-          },
+          attributes: utils.getSemanticAttributesFromConnection(this.options),
         });
 
+        span.setAttribute(
+          AttributeNames.IDLE_TIMEOUT_MILLIS,
+          this.options.idleTimeoutMillis
+        );
+        span.setAttribute(AttributeNames.MAX_CLIENT, this.options.maxClient);
```

Comment:
It is generally preferred to have all the attributes that are available set on `startSpan` so they can be used in samplers. I would reconsider setting them after span creation.

Reply:
Done